packages/proxy/lib/http/request-middleware.ts

```diff
@@ -283,19 +461,60 @@ const SendRequestOutgoing: RequestMiddleware = function () {
 
   const onSocketClose = () => {
     this.debug('request aborted')
+    // if the request is aborted, close out the middleware span and http span. the response middleware did not run
+
+    this.reqMiddlewareSpan?.setAttributes({
+      requestAborted: true,
+    })
+
+    this.reqMiddlewareSpan?.end()
+    this.handleHttpRequestSpan?.end()
+
     req.abort()
   }
 
   req.on('error', this.onError)
-  req.on('response', (incomingRes) => this.onResponse(incomingRes, req))
+  req.on('response', (incomingRes) => {
+    const { timings } = incomingRes.request
+
+    if (!timings.socket) {
```

Comment:
if we make the `time` option conditional, we might have to check for existence down here